src/java.base/share/classes/java/net/SocketPermission.java

```diff
@@ -1110,10 +1117,10 @@ public int hashCode() {
 
         }
 
-        if (invalid || cname == null)
+        if (invalid || cnames == null)
             return this.getName().hashCode();
         else
-            return this.cname.hashCode();
+            return this.cnames[0].hashCode();
```

Comment:
@jaysk1 Can this cause "equal" `SocketPermission` objects with the identical, multiple cname entries, but in a different order, to have unequal hash codes?